htmlproofer/plugin.py

```diff
@@ -146,7 +147,17 @@ def get_external_url(self, url, scheme, src_path):
     @lru_cache(maxsize=1000)
     def resolve_web_scheme(self, url: str) -> int:
         try:
-            response = self._session.get(url, timeout=URL_TIMEOUT)
+            headers = {}
+            max_bytes = self.config['read_max_bytes']
+            if max_bytes:
+                headers["Range"] = f"bytes=0-{max_bytes}"
+
+            response = self._session.get(url, timeout=URL_TIMEOUT, stream=True, headers=headers)
+
+            # Download the entire contents as to not break previous behaviour.
+            for _ in response.iter_content(chunk_size=1024):
+                pass
```

Comment:
If we don't actually care about downloading the contents (we seemingly don't do any verification of contents), then we could always just drop these lines, with `stream=True` set the body of the response isn't downloaded automatically.

Reply:
We don't as long as the url status is correct.

Reply:
Should I just simplify this MR to just skip the download entirely and drop the `read_max_bytes` configuration? I'm hesitant to break existing behaviour, so I'll list out a few possible routes this MR could go:
1. Keep this MR as it is, with `read_max_bytes` limiting size of downloaded content.
1. Opt for a different configuration value, such as `skip_downloads` which triggers this new behaviour of skipping the download of the responses body.
1. Don't bother adding any configurable behaviour and just establish a connection via `stream=True` and skip any downloads.
@manuzhang Which would be your preference?

Reply:
I'd prefer option 2.

Reply:
@manuzhang I've implemented option 2 and updated this MR, as well as the original issue title.